modules/bluefish/consumer/bluefish_consumer.cpp

```diff
@@ -56,10 +56,78 @@
 #include <array>
 
 namespace caspar { namespace bluefish { 
-			
+		
+enum class hardware_downstream_keyer_mode
+{
+	disable = 0,
+	enable = 1,
+};
+
+enum class hardware_downstream_keyer_audio_source
+{
+	SDIVideoInput = 1,
+	VideoOutputChannel = 2
+};
+
+enum class bluefish_hardware_output_channel
+{
+	// ** current Bluefish hardware only support 4224  on channels A and C.
+	hardware_4224_channel_a,
+	hardware_4224_channel_c,
+	hardware_422_channel_a,
+	hardware_422_channel_b,
+	hardware_422_channel_c,
+	hardware_422_channel_d,
+	default_output_channel = hardware_422_channel_a
+};
+
+EBlueVideoChannel get_bluesdk_videochannel_from_streamid(bluefish_hardware_output_channel streamid)
+{
+	/*This function would return the corresponding EBlueVideoChannel from the device output channel*/
+	switch (streamid)
+	{
+		case bluefish_hardware_output_channel::hardware_422_channel_a:	return BLUE_VIDEO_OUTPUT_CHANNEL_A;
+		case bluefish_hardware_output_channel::hardware_4224_channel_a:	return BLUE_VIDEO_OUTPUT_CHANNEL_A;
+		case bluefish_hardware_output_channel::hardware_422_channel_b:	return BLUE_VIDEO_OUTPUT_CHANNEL_B;
+		case bluefish_hardware_output_channel::hardware_4224_channel_c:	return BLUE_VIDEO_OUTPUT_CHANNEL_C;
+		case bluefish_hardware_output_channel::hardware_422_channel_c:	return BLUE_VIDEO_OUTPUT_CHANNEL_C;
+		case bluefish_hardware_output_channel::hardware_422_channel_d:	return BLUE_VIDEO_OUTPUT_CHANNEL_D;
+		default: return BLUE_VIDEO_OUTPUT_CHANNEL_A;
+	}
+}
+
+bool get_videooutput_channel_routing_info_from_streamid(bluefish_hardware_output_channel streamid,
+	EEpochRoutingElements & channelSrcElement,
+	EEpochRoutingElements & sdioutputDstElement)
+{
+	switch (streamid)
+	{
+	case bluefish_hardware_output_channel::hardware_4224_channel_a:	channelSrcElement = EPOCH_SRC_OUTPUT_MEM_INTERFACE_CHA;
```

Comment:
There should be a newline directly after `case bluefish_hardware_output_channel::hardware_4224_channel_a:`